synapse/storage/stream.py

```diff
@@ -95,6 +95,50 @@ def upper_bound(token, engine, inclusive=True):
         )
 
 
+def filter_to_clause(event_filter):
+    if not event_filter:
+        return "", []
+
+    clauses = []
+    args = []
+
+    if event_filter.types:
+        clauses.append(
+            "(%s)" % " OR ".join("type = ?" for _ in event_filter.types)
+        )
+        args.extend(event_filter.types)
```

Comment:
We currently allow wildcard matches for event types. https://github.com/matrix-org/synapse/blob/a98d2152049b0a61426ed3d8b6ac872a9ca3f535/synapse/api/filtering.py#L235